distributed/aws/s3.py

```diff
@@ -0,0 +1,34 @@
+import glob
+from pathlib import Path
+
+from logger import logger
+from .client import AWSClient
+
+
+class S3:
+    """Wrapper around boto3 to upload to/download from S3 bucket
+    """
+
+    def __init__(self, bucket: str):
+        self._client = AWSClient(service='s3').client
+        self._bucket = bucket
+
+    def add(self, path: str, key: str):
+        if not Path(path).exists():
+            logger.error(f'Invalid path: {path}! Nothing to upload!')
+            return
+        try:
+            logger.info(f'Uploading object from `{path}` to S3 bucket `{self._bucket}` key `{key}`')
+            for filename in glob.iglob(str(path) + '**/**', recursive=True):
+                if Path(filename).is_file():
+                    self._client.upload_file(filename, self._bucket, f'{key}/{filename}')
+        except Exception as exp:
+            logger.exception(f'Got following exception while uploading object to S3 \n{exp}')
+
+    def get(self, key: str, local_path: str = '.'):
+        try:
+            logger.info(f'Downloading object from `{self._bucket}:{key}` to local directory: {local_path}')
+            # for filename in glob.iglob(path + '**/**', recursive=True):
+            self._client.download_file(self._bucket, key, local_path)
+        except Exception as exp:
```

Comment:
We may want to raise those to the caller instead as it is unclear here what to do in these cases?

Reply:
Good point. Done